src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/MarkMembersAsStatic.cs

```diff
@@ -254,6 +255,17 @@ void OnSymbolEnd(SymbolAnalysisContext symbolEndContext)
                 return false;
             }
 
+            // Auto-properties (readonly, get, set) marked with an attribute have a block context callback while they don't
+            // if there is no attribute (see https://github.com/dotnet/roslyn/issues/46132).
+            // We consider that auto-property have the intent to always be instance members so we want to workaround this issue
+            // by bailing-out if this is an accessor, there are some operations in the block but they are all "none" operations.
+            if (methodSymbol.IsAccessorMethod()
+                && !blockOperations.IsEmpty
+                && !blockOperations.Where(op => op.Kind != OperationKind.None).Any())
+            {
+                return false;
+            }
```

Comment:
Will this break if https://github.com/dotnet/roslyn/issues/46132 is fixed?

Reply:
I am not 100% sure but IMO it shouldn't break. Current problem is that we don't have the callback for the other cases so:

- Solution 1: We no longer have any callback, then this logic becomes useless. I don't think Roslyn team will go in this direction.

- Solution 2: We now have callbacks also for auto-properties, I would assume the following snippet to remain valid and to also work for attributeless auto-properties.